fix: Add chain context on several more assetinput components that were missing it

evanaronson opened the pull request:
# Description

Several address input uses were missing the chain context and linked to the wrong block explorer. 

While I have already tested that this works in the contexts it fixed, this is obviously AI generated code, so maybe it's dumb.

## Type of Change

<!--- Please delete the options that are not relevant. -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Developer Checklist:

- [x] Manually smoke tested the functionality locally
- [x] Confirmed there are no new warnings or errors in the browser console
- [ ] (For User Stories only) Double-checked that all Acceptance Criteria are satisfied
- [x] Confirmed there are no new warnings on automated tests
- [x] Merged and published any dependent changes in downstream modules
- [x] Selected the correct base branch
- [ ] Commented the code in hard-to-understand areas
- [ ] Followed the code style guidelines of this project
- [ ] Reviewed that the Files Changed in Github’s UI reflect my intended changes
- [x] Confirmed the pipeline checks are not failing

## Review Checklist:

- [ ] Tested locally that all Acceptance Criteria or Expected Outcomes are satisfied
- [ ] Confirmed that changes follow the code style guidelines of this project

sepehr2github reviewed (changes requested):
Hey, thanks but I found a new issue, 
When I click on create dao the wrapping modal comes up with a loading! could you check that, Thanks

babarella commented:
> <img alt="Screenshot 2023-06-29 at 9 27 56 PM" width="372" src="https://user-images.githubusercontent.com/46657145/249858583-7afa615b-d674-42f0-a8c6-fd93c2af5809.png"> <img alt="Screenshot 2023-06-29 at 9 29 22 PM" width="409" src="https://user-images.githubusercontent.com/46657145/249858747-47d4f4c4-5776-4d07-b1f7-95a33595cedf.png">
> The description should have some variables in it, ANT should be replaced with dao token symbols,
> 
> Also, I couldn't wrap the token, When enter a number the button stay disabled, Also the mint token buttons should be hided as well if a underlying token exists <img alt="Screenshot 2023-06-29 at 9 30 25 PM" width="1188" src="https://user-images.githubusercontent.com/46657145/249858823-a9b5f734-433c-4e99-b8fa-dcd0f49ad5c8.png">

![image](https://github.com/aragon/app/assets/114730054/32fd26a8-64eb-438e-b593-ddaa5d5c5b8c)

Hey, Sepehr, I'm not sure what the issues on your side are but I am not able to reproduce them.
The token symbol is passed, just when daoToken is not loaded, then the placeholder is ANT for informational purposes.

sepehr2github reviewed (approved):
Thanks

evanaronson committed:
Update remaining addressinputs missing context

evanaronson committed:
Fix AddressInput links to correct block explorer

Fixed several instances of AddressInput that linked to the wrong block explorer.

Signed-off-by: Evan Aronson <93671071+evanaronson@users.noreply.github.com>

evanaronson commented on src/plugins/multisigPlugin/components/multisigSetupMembership/multisigSetupMembership.tsx:
```diff
@@ -1,15 +1,18 @@
+    const { formPrefix, disabled, onAddClick, pluginAddress, hideLabel, network, daoId } = props;
 
     const { t } = useTranslations();
+    const { data: dao } = useDao({ urlParams: { id: daoId ?? '' } }, { enabled: daoId != null });
```
I don't know what is happening here. Someone please review.

evanaronson commented on src/plugins/tokenPlugin/components/tokenSetupMembership/components/tokenSetupMembershipCreateToken/tokenSetupMembershipCreateToken.tsx:
```diff
@@ -22,10 +28,12 @@ const symbolMaxLength = 12;
     const { t } = useTranslations();
     const { setValue } = useFormContext();
+    const { data: dao } = useDao({ urlParams: { id: daoId } });
+    const chainId = dao ? networkDefinitions[dao.network].id : undefined;
```
I don't know what is happening here. Someone please review.

thekidnamedkd committed:
chore: impl useDaoChain

thekidnamedkd committed:
chore: refactor for useDaoChain

thekidnamedkd committed:
chore: linter, fix tests

evanaronson committed:
Merge branch 'main' into block-explorer-chain

thekidnamedkd committed:
chore: update changeset

milosh86 commented on src/shared/hooks/useDaoChain/useDaoChain.ts:
```diff
@@ -0,0 +1,26 @@
+
+    const shouldFetchDao = daoId != null && providedChainId == null && network == null;
+
+    const { data: dao, isLoading } = useDao({ urlParams: { id: daoId ?? '' } }, { enabled: shouldFetchDao });
```
You can get the network straight from `daoId`, without calling `useDao`, using:

`daoUtils.parseDaoId(daoId)`

milosh86 replied:
on the other side it could be useful to have full dao object here if you see some possible extensions in the future?

milosh86 reviewed (approved):
Nice one 👍 
Left one minor comment, but feel free to leave it as is if makes more sense, I left approve anyway.
